config/config.go

```diff
@@ -67,32 +66,92 @@ func (p *SigningProfile) populate(cfg *Config) bool {
 				p.Provider, err = auth.New(key.Key, nil)
 				if err != nil {
 					log.Debugf("failed to create new stanard auth provider: %v", err)
-					return false
+					return cferr.Wrap(cferr.PolicyError, cferr.InvalidPolicy,
+						errors.New("failed to create new stanard auth provider"))
 				}
 			} else {
 				log.Debugf("unknown authentication type %v", key.Type)
-				return false
+				return cferr.Wrap(cferr.PolicyError, cferr.InvalidPolicy,
+					errors.New("unknown authentication type"))
 			}
 		} else {
-			log.Debugf("failed to find auth_key %v in auth_keys section", p.AuthKeyName)
-			return false
+			return cferr.Wrap(cferr.PolicyError, cferr.InvalidPolicy,
+				errors.New("failed to find auth_key in auth_keys section"))
 		}
 	}
 
 	if p.RemoteName != "" {
 		if remote := cfg.Remotes[p.RemoteName]; remote != "" {
-			p.Remote = client.NewServer(remote)
-			if p.Remote == nil {
-				log.Debugf("failed to connect to remote %v", remote)
-				return false
+			if err := p.updateRemote(remote); err != nil {
+				return err
 			}
 		} else {
-			log.Debugf("failed to find remote %v in remotes section %v", p.RemoteName, cfg)
-			return false
+			return cferr.Wrap(cferr.PolicyError, cferr.InvalidPolicy,
+				errors.New("failed to find remote in remotes section"))
 		}
 	}
 
-	return true
+	return nil
+}
+
+// updateRemote takes a signing profile and initializes the remote server object
+// to the hostname:port combination sent by remote
+func (p *SigningProfile) updateRemote(remote string) error {
+	if remote != "" {
+		p.Remote = client.NewServer(remote)
+		if p.Remote == nil {
+			return cferr.Wrap(cferr.PolicyError, cferr.InvalidRequest,
+				errors.New("failed to connect to remote"))
+		}
+	}
+	return nil
+}
+
+// OverrideRemotes takes a signing configuration and updates the remote server object
+// to the hostname:port combination sent by remote
+func (p *Signing) OverrideRemotes(remote string) error {
```

Comment:
SigningProfile is a profile and Signing is a policy, we're kind of stuck for single or double letter names. Let's stick with p for now for everything in this file.